src/util/sss_python.h

```diff
@@ -39,9 +49,12 @@ sss_exception_with_doc(const char *name, const char *doc, PyObject *base,
         MODINITERROR;                               \
```

Comment:
But here we need to dec module reference as well, right?

Reply:
If I understand it correctly the dereference only applies when some constant is added to the module. In this case [PyType_Ready](https://docs.python.org/3/c-api/type.html#c.PyType_Ready) finishes the initialization of the object, there isn't any constant value added.
Moreover, taking a look at the [example page](https://github.com/python/cpython/blob/5c7ed7550ec2da16d7679e538fcd7c1a5631811f/Doc/includes/custom.c#L30), there isn't any dereference.

Reply:
IIUC one should dec ref counter if module is created but init function doesn't return it.

The difference with example you refer to: module isn't created to this moment in the example.

And, btw, take a look where this macro is used - for example `initpyhbac()` - a lot of stuff is already added to the module. But as I said, that doesn't matter.